waves/project.py

```diff
@@ -2846,3 +2884,402 @@ def generate_report_lcoe_breakdown(self) -> pd.DataFrame:
+            "windspeed_{number}m", where `{number}` represents the sensor height in meters.
+
+        Returns
+        -------
+        dict[str, int]
+            A dictionary containing the first two columns that match the "windspeed_{number}m" pattern.
+            The keys are the column names (e.g., "windspeed_10m"), and the values are the corresponding sensor heights (e.g., 10).
+        """
+        windspeed_columns = {}
+    
+        # Regex pattern to match "windspeed_{number}m" where the name ends with 'm'
+        pattern = r"windspeed_(\d+)m$"
+    
+        for col in df.columns:
+            match = re.match(pattern, col)
+            if match:
+                number = match.group(1)
+                windspeed_columns[col] = int(number)
+    
+        return dict(list(windspeed_columns.items())[:2])
+
+
+    def fit_weibull_distribution(self, height, random_seed=1):
+        """Fits a Weibull distribution to wind speed data at a specified height.
+
+        This function fits a Weibull distribution to the wind speed data at the specified height 
+        from the weather data. It assumes that wind speeds are non-negative, so it fixes the 
+        location parameter of the Weibull distribution to 0. If the required column for the given 
+        height is not present in the weather data, an error message is printed and the function 
+        returns a string indicating the missing data.
+
+        Parameters
+        ----------
+        height : int
+            The height (in meters) at which the wind speed data is collected. The function looks for 
+            a column named "windspeed_{height}m" in the weather data.
+        random_seed : int, optional
+            A random seed for reproducibility. Defaults to 1. This seed is used to initialize the random 
+            number generator for the Weibull fitting procedure.
+
+        Returns
+        -------
+        float | str
+            If the wind speed data is available for the specified height, the function returns the shape 
+            parameter of the fitted Weibull distribution. If the necessary column is missing from the weather 
+            data, it returns a string indicating the error (e.g., "wind speed at {height}m not provided").
+        """
+        # Set random seed for reproducibility
+        np.random.seed(random_seed)
+
+        windspeed_column = "windspeed_" + str(height) + "m"
+
+        df = self.wombat.env.weather.to_pandas()
+    
+        # Check if the necessary columns are in the dataframe
+        if windspeed_column not in df.columns:
+            print("wind speed at " + str(height) + "m not provided at " + str(self.library_path) + "\\weather\\" + str(self.weather_profile) + "\nPlease, add a column to the weather .csv file with the name 'windspeed_" + str(height) + "m', and the respective wind speed data")
+            return "wind speed at " + str(height) + "m not provided"
```

Comment:
This should be an error message like the one above, and using the same formatting approach.